src/shared/components/ncTable/partials/TableHeaderColumnOptions.vue

```diff
@@ -0,0 +1,238 @@
+				v-if="getPossibleMagicFields(column).length > 0 && !canFilterWithTextInput"
+				:title="t('tables', 'Choose value')" />
+			<NcActionButton
+				v-for="(magicField, index) in getMagicFields"
+				:key="'magic-field-' + index"
+				:value="magicField.id"
+				:checked="index === 0"
+				:icon="magicField.icon"
+				@click="submitMagicField(magicField.id)">
+				{{ magicField.label }}
+			</NcActionButton>
+		</NcActions>
+	</div>
+</template>
+
+<script>
+import textLineMixin from '../mixins/columnsTypes/textLineMixin.js'
+import textLinkMixin from '../mixins/columnsTypes/textLinkMixin.js'
+import selectionMixin from '../mixins/columnsTypes/selectionMixin.js'
+import numberMixin from '../mixins/columnsTypes/numberMixin.js'
+import selectionCheckMixin from '../mixins/columnsTypes/selectionCheckMixin.js'
+import numberStarsMixin from '../mixins/columnsTypes/numberStarsMixin.js'
+import numberProgressMixin from '../mixins/columnsTypes/numberProgressMixin.js'
+import datetimeDateMixin from '../mixins/columnsTypes/datetimeDateMixin.js'
+import datetimeTimeMixin from '../mixins/columnsTypes/datetimeTimeMixin.js'
+import datetimeMixin from '../mixins/columnsTypes/datetimeMixin.js'
+import generalHelper from '../../../mixins/generalHelper.js'
+import SortAsc from 'vue-material-design-icons/SortAscending.vue'
+import SortDesc from 'vue-material-design-icons/SortDescending.vue'
+import Pencil from 'vue-material-design-icons/Pencil.vue'
+import { NcActions, NcActionButton, NcActionInput, NcActionButtonGroup, NcActionSeparator, NcActionCaption, NcActionRadio } from '@nextcloud/vue'
+import { mapState } from 'vuex'
+import searchAndFilterMixin from '../mixins/searchAndFilterMixin.js'
+
+export default {
+
+	components: {
+		Pencil,
+		NcActionInput,
+		NcActionRadio,
+		NcActionCaption,
+		NcActionButton,
+		NcActions,
+		SortAsc,
+		SortDesc,
+		NcActionButtonGroup,
+		NcActionSeparator,
+	},
+
+	mixins: [
+		textLineMixin,
+		selectionMixin,
+		numberMixin,
+		generalHelper,
+		selectionCheckMixin,
+		textLinkMixin,
+		numberStarsMixin,
+		numberProgressMixin,
+		datetimeDateMixin,
+		datetimeTimeMixin,
+		datetimeMixin,
+		searchAndFilterMixin,
+	],
+
+	props: {
+		column: {
+		      type: Object,
+		      default: null,
+		    },
+		openState: {
+		      type: Boolean,
+		      default: false,
+		    },
+	},
+
+	data() {
+		return {
+			filterValue: '',
+			operator: '',
+		}
+	},
+
+	computed: {
+		...mapState({
+			view: state => state.data.view,
+		}),
+		getOperators() {
+			console.debug('getOperators requested')
+			const possibleOperators = this.getPossibleOperators(this.column)
+
+			// preselect first operator, even if it's not displayed
+			if (this.operator === '') {
+				console.debug('operator is empty, try to set first option', possibleOperators)
+				// eslint-disable-next-line vue/no-side-effects-in-computed-properties
+				this.operator = possibleOperators[0]?.id ?? ''
+				console.debug('operator', this.operator)
+			}
```

Comment:
I would not expect a change to a data value inside a computed.

It's probably fine as there's only few triggers to `getOperators` and setting a default should be fine almost anytime.
However I think it would be more clean to have a separate computed for example `operatorWithPreselect` that returns `this.operator || possibleOperators[0].id ?? ''` or so.